crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -839,3 +839,11 @@ fn prefix_operand_type_rules(op: &crate::UnaryOp, rhs_type: &Type) -> Result<Typ
     }
     Ok(rhs_type.clone())
 }
+
```

Comment:
I'll leave whether we need to get rid of the clone to @jfecher .

For this function, we should link to: https://stackoverflow.com/a/47127500
So there is a reference to where it came from